### PR TITLE
Minor enhancements and fixes for OSDP

### LINKED ISSUE #28168: [Coverity CID :214218] Unused value in subsys/mgmt/osdp/src/osdp.c
Static code scan issues found in file:

https://github.com/zephyrproject-rtos/zephyr/tree/27e1fd69f987d3082229b3368a87d59ffdcab5a4/subsys/mgmt/osdp/src/osdp.c#L62

Category: Code maintainability issues
Function: `osdp_uart_isr`
Component: Other
CID: [214218](https://scan9.coverity.com/reports.htm#v29726/p12996/mergedDefectId=214218)

Details:

```
56    
57             if (uart_irq_tx_ready(dev)) {
58                 len = ring_buf_get(&p->tx_buf, buf, 1);
59                 if (!len) {
60                     uart_irq_tx_disable(dev);
61                 } else {
>>>     CID 214218:  Code maintainability issues  (UNUSED_VALUE)
>>>     Assigning value from "uart_fifo_fill(dev, buf, 1)" to "wrote" here, but that stored value is overwritten before it can be used.
62                     wrote = uart_fifo_fill(dev, buf, 1);
63                 }
64             }
65         }
66     #ifdef CONFIG_OSDP_MODE_PD
67         /* wake osdp_refresh thread */
```

Please fix or provide comments in coverity using the link:

https://scan9.coverity.com/reports.htm#v32951/p12996.

Note: This issue was created automatically. Priority was set based on classification
of the file affected and the impact field in coverity. Assignees were set using the CODEOWNERS file.

### PR DESCRIPTION
**Pull request by sidcha**
These patches were part of #28200 creating a separate PR for fixes and enhancements so that PR can remain on topic.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by cbsiddharth** [comment]
@carlescufi review please!

File: samples/subsys/mgmt/osdp/peripheral_device/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2020 Siddharth Chandrasekaran <siddharth@embedjournal.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+cmake_minimum_required(VERSION 3.13.1)
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
```
Comment (carlescufi) [inline samples/subsys/mgmt/osdp/peripheral_device/CMakeLists.txt:8]: This should be:
```
find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
```

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: samples/subsys/mgmt/osdp/peripheral_device/CMakeLists.txt (hunk 1 above)
Comment (cbsiddharth) [reply samples/subsys/mgmt/osdp/peripheral_device/CMakeLists.txt]: Done.

**Comment by cbsiddharth** [comment]
@carlescufi I've addressed your comment. Please check again.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**